buildSrc/src/main/java/org/elasticsearch/gradle/tool/DockerUtils.java

```diff
@@ -208,14 +208,26 @@ private static Result runCommand(Project project, String... args) {
         ByteArrayOutputStream stdout = new ByteArrayOutputStream();
         ByteArrayOutputStream stderr = new ByteArrayOutputStream();
 
-        final ExecResult execResult = project.exec(spec -> {
-            // The redundant cast is to silence a compiler warning.
-            spec.setCommandLine((Object[]) args);
-            spec.setStandardOutput(stdout);
-            spec.setErrorOutput(stderr);
-        });
-
-        return new Result(execResult.getExitValue(), stdout.toString(), stderr.toString());
+        try {
+            final ExecResult execResult = project.exec(spec -> {
+                // The redundant cast is to silence a compiler warning.
+                spec.setCommandLine((Object[]) args);
+                spec.setStandardOutput(stdout);
+                spec.setErrorOutput(stderr);
+            });
+            return new Result(execResult.getExitValue(), stdout.toString(), stderr.toString());
+        } catch (GradleException ex) {
```

Comment:
Rather than doing this via generic exception handling let's set `spec.setIgnoreExitValue(true)` and throw our own exception if the result is a non-zero exit value. That way if we blow up for some other reason (bad config, Gradle issue) we aren't returning a potentially misleading exception message.